components/SQLEditor.go

```diff
@@ -81,3 +87,99 @@ func (s *SQLEditor) SetBlur() {
 	s.SetBorderColor(tcell.ColorWhite)
 	s.SetTextStyle(tcell.StyleDefault.Foreground(tcell.ColorWhite))
 }
+
+
+/*
+	THIS FUNCTION OPENS EXTERNAL EDITOR.
+
+	Q: WHY OPEN ANOTHER TERMINAL?
+	A: OPENING EDITORS LIKE VIM/NEOVIM REALLY MESSED UP INITIAL TERMINAL'S OUTPUT.
+*/
+
+func openExternalEditor(s *SQLEditor) string {
+	// Current folder as path of temporary file
+	path := "./lazysql.sql"
+
+	editor := getEditor()
+	terminal := getTerminal()
+
+	// Create a temporary file with the current SQL query content
+	content := []byte(s.TextArea.GetText())
+
+	/*
+	0644 Permission
+	* User: read & write
+	* Group: read
+	* Other: read
+	*/
+
+	err := os.WriteFile(path, content, 0644)
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Remove the temporary file with the end of function
+	defer os.Remove(path)
+
+	// Setup command
+	cmd := exec.Command(terminal, "-e", editor, path)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+
+	// cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Read the updated content from the temporary file
+	updatedContent, err := os.ReadFile(path)
+	if err != nil {
+		return s.TextArea.GetText()
+	}
+
+	// Convert to string before returning
+	return string(updatedContent)
+}
+
+// Function to select editor
+func getEditor() string {
+	var editor string = os.Getenv("SQL_EDITOR")
+	if editor == "" {
+		editor = os.Getenv("EDITOR")
+	}
+
+	if editor == "" {
+		editor = os.Getenv("VISUAL")
+	}
+
+	if editor == "" {
+		editor = "vi" 		// use "vi" if $EDITOR not set
+	}
+
+	return editor
+}
+
+// Function to select terminal
+func getTerminal() string {
+	var terminal string = os.Getenv("SQL_TERMINAL")
+
+	if terminal == "" {
+		terminal = os.Getenv("TERMINAL")
+	}
+	
+	// Check if x-terminal-emulator exists.
+	x_term_emu, err := exec.LookPath("x-terminal-emulator")
```

Comment:
Each time a snake case is used in Go, a kitten dies

Reply:
Thought would be easier to understand and work with. 

Should I change it to camel case?

Reply:
I would say call it `terminalEmulator` initializes with xterm and overloads it if xterm in all emulator look up finds it

Reply:
Changed it accordingly.

Reply:
Did you push on our changes?

Reply:
Yes.